hw/ip/otbn/dv/model/iss_wrapper.cc

```diff
@@ -218,6 +219,17 @@ static uint32_t read_ext_reg(const std::string &reg_name,
   return val;
 }
 
+static std::string wlen_val_to_hex_str(uint32_t val[8]) {
+  std::ostringstream oss;
+
+  oss << std::hex << "0x";
+  for (int i = 7; i >= 0; --i) {
+    oss << std::setfill('0') << std::setw(8) << val[i];
+  }
+
+  return oss.str();
+}
```

Comment:
Honestly? I'd have written this it the much more "dumb" way as

```
std::string out(2 + 8 * 8, 0);
std::sprintf(out.data(), "0x%08x%08x%08x%08x%08x%08x%08x%08x", val[0], va[1], ...);
return out;
```
It avoids fussing with C++ streams which are kind of a mess. (In general, I think std:;sprintf is the way to go...)

Reply:
Using the stream formatter stuff (setfill etc) is a little awkward but this isn't a complex use case and I think it's pretty obvious what's going on. Plus everything else is using C++ streams so I think a single use of std::sprintf wouldn't fit well here.